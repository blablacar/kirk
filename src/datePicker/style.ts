// DO NOT EDIT
// This CSS contents is extracted from 'react-dates/lib/css/_datepicker.css'
import css from 'styled-jsx/css'

export default css`
  :global(.SingleDatePicker) {
    position: relative;
    display: inline-block;
  }
  :global(.SingleDatePicker__block) {
    display: block;
  }
  :global(.SingleDatePicker_picker) {
    z-index: 1;
    background-color: #fff;
    position: absolute;
  }
  :global(.SingleDatePicker_picker__rtl) {
    direction: rtl;
  }
  :global(.SingleDatePicker_picker__directionLeft) {
    left: 0;
  }
  :global(.SingleDatePicker_picker__directionRight) {
    right: 0;
  }
  :global(.SingleDatePicker_closeButton) {
    background: 0 0;
    border: 0;
    color: inherit;
    font: inherit;
    line-height: normal;
    overflow: visible;
    cursor: pointer;
    position: absolute;
    top: 0;
    right: 0;
    padding: 15px;
    z-index: 2;
  }
  :global(.SingleDatePicker_closeButton:focus),
  :global(.SingleDatePicker_closeButton:hover) {
    color: darken(#cacccd, 10%);
    text-decoration: none;
  }
  :global(.SingleDatePicker_closeButton_svg) {
    height: 15px;
    width: 15px;
    fill: #cacccd;
  }
  :global(.DayPicker) {
    background: #fff;
    position: relative;
    text-align: left;
  }
  :global(.DayPicker__horizontal) {
    background: #fff;
  }
  :global(.DayPicker__verticalScrollable) {
    height: 100%;
  }
  :global(.DayPicker__hidden) {
    visibility: hidden;
  }
  :global(.DayPicker__withBorder) {
    box-shadow: 0 2px 6px rgba(0, 0, 0, 0.05), 0 0 0 1px rgba(0, 0, 0, 0.07);
    border-radius: 3px;
  }
  :global(.DayPicker_portal__horizontal) {
    box-shadow: none;
    position: absolute;
    left: 50%;
    top: 50%;
  }
  :global(.DayPicker_portal__vertical) {
    position: initial;
  }
  :global(.DayPicker_focusRegion) {
    outline: 0;
  }
  :global(.DayPicker_weekHeaders) {
    position: relative;
  }
  :global(.DayPicker_weekHeaders__horizontal) {
    margin-left: 9px;
  }
  :global(.DayPicker_weekHeader) {
    color: #757575;
    position: absolute;
    top: 62px;
    z-index: 2;
    padding: 0 13px;
    text-align: left;
  }
  :global(.DayPicker_weekHeader__vertical) {
    left: 50%;
  }
  :global(.DayPicker_weekHeader__verticalScrollable) {
    top: 0;
    display: table-row;
    border-bottom: 1px solid #dbdbdb;
    background: #fff;
    margin-left: 0;
    left: 0;
    width: 100%;
    text-align: center;
  }
  :global(.DayPicker_weekHeader_ul) {
    list-style: none;
    margin: 1px 0;
    padding-left: 0;
    padding-right: 0;
    font-size: 14px;
  }
  :global(.DayPicker_weekHeader_li) {
    display: inline-block;
    text-align: center;
  }
  :global(.DayPicker_transitionContainer) {
    position: relative;
    overflow: hidden;
    border-radius: 3px;
  }
  :global(.DayPicker_transitionContainer__horizontal) {
    transition: height 0.2s ease-in-out;
  }
  :global(.DayPicker_transitionContainer__vertical) {
    width: 100%;
  }
  :global(.DayPicker_transitionContainer__verticalScrollable) {
    padding-top: 20px;
    height: 100%;
    position: absolute;
    top: 0;
    bottom: 0;
    right: 0;
    left: 0;
    overflow-y: scroll;
  }
  :global(.DayPickerKeyboardShortcuts_buttonReset) {
    background: 0 0;
    border: 0;
    border-radius: 0;
    color: inherit;
    font: inherit;
    line-height: normal;
    overflow: visible;
    padding: 0;
    cursor: pointer;
    font-size: 14px;
  }
  :global(.DayPickerKeyboardShortcuts_buttonReset:active) {
    outline: 0;
  }
  :global(.DayPickerKeyboardShortcuts_show) {
    width: 22px;
    position: absolute;
    z-index: 2;
  }
  :global(.DayPickerKeyboardShortcuts_show__bottomRight) {
    border-top: 26px solid transparent;
    border-right: 33px solid #00a699;
    bottom: 0;
    right: 0;
  }
  :global(.DayPickerKeyboardShortcuts_show__bottomRight:hover) {
    border-right: 33px solid #008489;
  }
  :global(.DayPickerKeyboardShortcuts_show__topRight) {
    border-bottom: 26px solid transparent;
    border-right: 33px solid #00a699;
    top: 0;
    right: 0;
  }
  :global(.DayPickerKeyboardShortcuts_show__topRight:hover) {
    border-right: 33px solid #008489;
  }
  :global(.DayPickerKeyboardShortcuts_show__topLeft) {
    border-bottom: 26px solid transparent;
    border-left: 33px solid #00a699;
    top: 0;
    left: 0;
  }
  :global(.DayPickerKeyboardShortcuts_show__topLeft:hover) {
    border-left: 33px solid #008489;
  }
  :global(.DayPickerKeyboardShortcuts_showSpan) {
    color: #fff;
    position: absolute;
  }
  :global(.DayPickerKeyboardShortcuts_showSpan__bottomRight) {
    bottom: 0;
    right: -28px;
  }
  :global(.DayPickerKeyboardShortcuts_showSpan__topRight) {
    top: 1px;
    right: -28px;
  }
  :global(.DayPickerKeyboardShortcuts_showSpan__topLeft) {
    top: 1px;
    left: -28px;
  }
  :global(.DayPickerKeyboardShortcuts_panel) {
    overflow: auto;
    background: #fff;
    border: 1px solid #dbdbdb;
    border-radius: 2px;
    position: absolute;
    top: 0;
    bottom: 0;
    right: 0;
    left: 0;
    z-index: 2;
    padding: 22px;
    margin: 33px;
  }
  :global(.DayPickerKeyboardShortcuts_title) {
    font-size: 16px;
    font-weight: 700;
    margin: 0;
  }
  :global(.DayPickerKeyboardShortcuts_list) {
    list-style: none;
    padding: 0;
    font-size: 14px;
  }
  :global(.DayPickerKeyboardShortcuts_close) {
    position: absolute;
    right: 22px;
    top: 22px;
    z-index: 2;
  }
  :global(.DayPickerKeyboardShortcuts_close:active) {
    outline: 0;
  }
  :global(.DayPickerKeyboardShortcuts_closeSvg) {
    height: 15px;
    width: 15px;
    fill: #cacccd;
  }
  :global(.DayPickerKeyboardShortcuts_closeSvg:focus),
  :global(.DayPickerKeyboardShortcuts_closeSvg:hover) {
    fill: #82888a;
  }
  :global(.KeyboardShortcutRow) {
    list-style: none;
    margin: 6px 0;
  }
  :global(.KeyboardShortcutRow__block) {
    margin-bottom: 16px;
  }
  :global(.KeyboardShortcutRow_keyContainer) {
    display: inline-block;
    white-space: nowrap;
    text-align: right;
    margin-right: 6px;
  }
  :global(.KeyboardShortcutRow_keyContainer__block) {
    text-align: left;
    display: inline;
  }
  :global(.KeyboardShortcutRow_key) {
    font-family: monospace;
    font-size: 12px;
    text-transform: uppercase;
    background: #f2f2f2;
    padding: 2px 6px;
  }
  :global(.KeyboardShortcutRow_action) {
    display: inline;
    word-break: break-word;
    margin-left: 8px;
  }
  :global(.DayPickerNavigation_container) {
    position: relative;
    z-index: 2;
  }
  :global(.DayPickerNavigation_container__vertical) {
    background: #fff;
    box-shadow: 0 0 5px 2px rgba(0, 0, 0, 0.1);
    position: absolute;
    bottom: 0;
    left: 0;
    height: 52px;
    width: 100%;
  }
  :global(.DayPickerNavigation_container__verticalScrollable) {
    position: relative;
  }
  :global(.DayPickerNavigation_button) {
    cursor: pointer;
    line-height: 0.78;
    user-select: none;
  }
  :global(.DayPickerNavigation_button__default) {
    border: 1px solid #e4e7e7;
    background-color: #fff;
    color: #757575;
  }
  :global(.DayPickerNavigation_button__default:focus),
  :global(.DayPickerNavigation_button__default:hover) {
    border: 1px solid #c4c4c4;
  }
  :global(.DayPickerNavigation_button__default:active) {
    background: #f2f2f2;
  }
  :global(.DayPickerNavigation_button__horizontal) {
    border-radius: 3px;
    padding: 6px 9px;
    top: 18px;
    position: absolute;
  }
  :global(.DayPickerNavigation_leftButton__horizontal) {
    left: 22px;
  }
  :global(.DayPickerNavigation_rightButton__horizontal) {
    right: 22px;
  }
  :global(.DayPickerNavigation_button__vertical) {
    display: inline-block;
    position: relative;
    height: 100%;
    width: 50%;
  }
  :global(.DayPickerNavigation_button__vertical__default) {
    padding: 5px;
  }
  :global(.DayPickerNavigation_nextButton__vertical__default) {
    border-left: 0;
  }
  :global(.DayPickerNavigation_nextButton__verticalScrollable) {
    width: 100%;
  }
  :global(.DayPickerNavigation_svg__horizontal) {
    height: 19px;
    width: 19px;
    fill: #82888a;
  }
  :global(.DayPickerNavigation_svg__vertical) {
    height: 42px;
    width: 42px;
    fill: #565a5c;
  }
  :global(.CalendarMonthGrid) {
    background: #fff;
    text-align: left;
    z-index: 0;
  }
  :global(.CalendarMonthGrid__animating) {
    z-index: 1;
  }
  :global(.CalendarMonthGrid__horizontal) {
    position: absolute;
    left: 9px;
  }
  :global(.CalendarMonthGrid__vertical) {
    margin: 0 auto;
  }
  :global(.CalendarMonthGrid__vertical_scrollable) {
    margin: 0 auto;
    overflow-y: scroll;
  }
  :global(.CalendarMonthGrid_month__horizontal) {
    display: inline-block;
    vertical-align: top;
    min-height: 100%;
  }
  :global(.CalendarMonthGrid_month__hideForAnimation) {
    position: absolute;
    z-index: -1;
    opacity: 0;
    pointer-events: none;
  }
  :global(.CalendarMonthGrid_month__hidden) {
    visibility: hidden;
  }
  :global(.CalendarMonth) {
    background: #fff;
    text-align: center;
    padding: 0 13px;
    vertical-align: top;
    user-select: none;
  }
  :global(.CalendarMonth_table) {
    border-collapse: collapse;
    border-spacing: 0;
  }
  :global(.CalendarMonth_caption) {
    color: #565a5c;
    font-size: 18px;
    text-align: center;
    padding-top: 22px;
    padding-bottom: 37px;
    caption-side: initial;
  }
  :global(.CalendarMonth_caption__verticalScrollable) {
    padding-top: 12px;
    padding-bottom: 7px;
  }
  :global(.CalendarDay) {
    box-sizing: border-box;
    cursor: pointer;
    font-size: 14px;
    text-align: center;
  }
  :global(.CalendarDay:active) {
    outline: 0;
  }
  :global(.CalendarDay__defaultCursor) {
    cursor: default;
  }
  :global(.CalendarDay__default) {
    border: 1px solid #e4e7e7;
    color: #565a5c;
    background: #fff;
  }
  :global(.CalendarDay__default:hover) {
    background: #e4e7e7;
    border: 1px double #e4e7e7;
    color: inherit;
  }
  :global(.CalendarDay__hovered_offset) {
    background: #f4f5f5;
    border: 1px double #e4e7e7;
    color: inherit;
  }
  :global(.CalendarDay__outside) {
    border: 0;
    background: #fff;
    color: #565a5c;
  }
  :global(.CalendarDay__blocked_minimum_nights) {
    background: #fff;
    border: 1px solid #eceeee;
    color: #cacccd;
  }
  :global(.CalendarDay__blocked_minimum_nights:active),
  :global(.CalendarDay__blocked_minimum_nights:hover) {
    background: #fff;
    color: #cacccd;
  }
  :global(.CalendarDay__highlighted_calendar) {
    background: #ffe8bc;
    color: #565a5c;
  }
  :global(.CalendarDay__highlighted_calendar:active),
  :global(.CalendarDay__highlighted_calendar:hover) {
    background: #ffce71;
    color: #565a5c;
  }
  :global(.CalendarDay__selected_span) {
    background: #66e2da;
    border: 1px solid #33dacd;
    color: #fff;
  }
  :global(.CalendarDay__selected_span:active),
  :global(.CalendarDay__selected_span:hover) {
    background: #33dacd;
    border: 1px solid #33dacd;
    color: #fff;
  }
  :global(.CalendarDay__last_in_range) {
    border-right: #00a699;
  }
  :global(.CalendarDay__selected),
  :global(.CalendarDay__selected:active),
  :global(.CalendarDay__selected:hover) {
    background: #00a699;
    border: 1px solid #00a699;
    color: #fff;
  }
  :global(.CalendarDay__hovered_span),
  :global(.CalendarDay__hovered_span:hover) {
    background: #b2f1ec;
    border: 1px solid #80e8e0;
    color: #007a87;
  }
  :global(.CalendarDay__hovered_span:active) {
    background: #80e8e0;
    border: 1px solid #80e8e0;
    color: #007a87;
  }
  :global(.CalendarDay__blocked_calendar),
  :global(.CalendarDay__blocked_calendar:active),
  :global(.CalendarDay__blocked_calendar:hover) {
    background: #cacccd;
    border: 1px solid #cacccd;
    color: #82888a;
  }
  :global(.CalendarDay__blocked_out_of_range),
  :global(.CalendarDay__blocked_out_of_range:active),
  :global(.CalendarDay__blocked_out_of_range:hover) {
    background: #fff;
    border: 1px solid #e4e7e7;
    color: #cacccd;
  }
`
