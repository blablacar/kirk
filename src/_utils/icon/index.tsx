import React from 'react'
import cc from 'classcat'
import isEmpty from 'lodash.isempty'

import Badge from 'badge'
import style from 'icon/style'
import { color } from '_utils/branding'

export interface IconProps extends Icon {
  readonly children: JSX.Element
  readonly viewBox?: string
}

const BaseIcon = ({
  className,
  iconColor,
  size,
  title,
  children,
  viewBox = '0 0 24 24',
  badgeContent,
  badgeAriaLabel,
}: IconProps) => {
  const icon = (
    <svg
      viewBox={viewBox}
      xmlns="http://www.w3.org/2000/svg"
      className={cc(['kirk-icon', className])}
      width={size}
      height={size}
      aria-hidden={isEmpty(title)}
      fill={iconColor}
    >
      {title && <title>{title}</title>}
      {children}
    </svg>
  )

  if (badgeContent) {
    return (
      <div className="kirk-icon-wrapper">
        {icon}
        <Badge className="kirk-icon-badge" ariaLabel={badgeAriaLabel}>
          {badgeContent}
        </Badge>
        <style jsx>{style}</style>
      </div>
    )
  }

  return icon
}

BaseIcon.defaultProps = {
  className: '',
  iconColor: color.icon,
  size: 24,
  title: '',
  badgeAriaLabel: '',
  badgeContent: '',
}

export default BaseIcon
