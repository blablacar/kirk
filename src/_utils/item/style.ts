import css from 'styled-jsx/css'

import { color, space } from '_utils/branding'

export default css`
  :global(.kirk-item) {
    position: relative;
    display: flex;
    padding: 0;
    padding-top: ${space.l};
    padding-bottom: ${space.l};
    align-items: center;
    flex: 1;
    border: 0;
    background-color: ${color.white}
  }

  :global(a.kirk-item) {
    cursor: pointer;
    background: none;
    text-decoration: none;
    user-select: none;
    -webkit-tap-highlight-color: ${color.tapHighlight};
    -webkit-touch-callout: none;
  }

  /* Button tag fixes */
  :global(button.kirk-item) {
    cursor: pointer;
    text-align: left;
    width: 100%;
    font-family: inherit;
    outline: none;
    -webkit-tap-highlight-color: ${color.tapHighlight};
    -webkit-touch-callout: none;
  }

  /* Text areas */
  .kirk-item-leftText {
    flex: 1;
  }

  .kirk-item-rightText {
    margin-left: ${space.l};
    text-align: right;
  }

  /* Text */
  :global(.kirk-item-title) {
    margin-bottom: ${space.s};
  }

  /* Addons */
  :global(.kirk-item-leftAddon),
  :global(.kirk-item-rightAddon),
  :global(kirk-item-chevron) {
    display: inline-flex;
    min-width: 24px;
    align-items: center;
  }

  :global(.kirk-item-leftAddon) {
    margin-right: ${space.l};
  }

  :global(.kirk-item-rightAddon),
  :global(kirk-item-chevron) {
    margin-left: ${space.l};
  }
`
