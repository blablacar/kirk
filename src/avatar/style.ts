import css from 'styled-jsx/css'
import { color } from '_utils/branding'

export default css`
  div {
    border: 2px dashed ${color.border};
    box-sizing: border-box;
    border-radius: 50%;
    position: relative;
    width: 48px;
    height: 48px;
  }

  .kirk-medium {
    height: 96px;
    width: 96px;
  }

  .kirk-large {
    height: 156px;
    width: 156px;
  }

  .kirk-image {
    border: none;
  }

  img {
    border-radius: 50%;
    width: 100%;
    height: 100%;
    object-fit: cover;
  }

  .kirk-idCheck {
    bottom: -3px;
    right: -3px;
    position: absolute;
    width: 18px;
    height: 18px;
    border-radius: 50%;
    background: ${color.success};
  }

  .kirk-medium .kirk-idCheck,
  .kirk-large .kirk-idCheck {
    bottom: 0;
    right: 0;
  }

  .kirk-medium .kirk-idCheck {
    width: 24px;
    height: 24px;
  }

  .kirk-large .kirk-idCheck {
    width: 36px;
    height: 36px;
  }
`
