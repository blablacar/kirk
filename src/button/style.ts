import css from 'styled-jsx/css'
import { transparentize } from 'polished'
import { color, font, space, transition, buttonIconSize, shadow } from '_utils/branding'

const borderSize = '2px'

export default css`
  :global(.kirk-button) {
    position: relative;
    box-sizing: border-box;
    display: inline-block;
    padding: ${space.l} ${space.xl};
    border: 1px solid transparent;
    border-radius: 48px;
    font-size: ${font.base.size};
    line-height: 1;
    text-align: center;
    text-decoration: none;
    cursor: pointer;
    font-family: inherit;
    min-height: ${buttonIconSize};
    min-width: ${buttonIconSize};
    max-width: 100%;
    overflow: hidden;
    user-select: none;
    transition: max-width ${transition.duration.fast} ease-in,
      background-color ${transition.duration.base} ease-in;
  }

  :global(.kirk-button.kirk-itemChoice) {
    border-radius: 0;
    overflow: visible;
    width: 100%;
  }

  :global(.kirk-button span) {
    position: relative;
    display: inline-block;
    width: auto;
    white-space: nowrap;
    text-overflow: ellipsis;
  }

  :global(.kirk-button:hover),
  :global(.kirk-button:focus),
  :global(.kirk-button:active) {
    text-decoration: none;
    outline: 0;
  }

  :global(.kirk-button-unstyled) {
    background-color: transparent;
    border-radius: inherit;
    color: ${color.accent};
    padding: inherit;
    display: inherit;
    line-height: inherit;
    font-size: inherit;
    text-align: left;
    min-width: inherit;
    min-height: inherit;
  }

  :global(.kirk-button[disabled]) {
    opacity: 0.5;
    cursor: default;
  }

  /* for unstyled disabled buttons, no opacity decrease (mainly icon buttons) */
  :global(.kirk-button-unstyled[disabled]) {
    opacity: 1;
  }

  :global(.kirk-button-primary) {
    background-color: ${color.primary};
    color: ${color.white};
  }

  :global(.kirk-button-shadowed) {
    border: ${borderSize} solid ${color.white};
    box-shadow: ${shadow.default};
    box-sizing: border-box;
  }

  :global(.kirk-button-primary:hover),
  :global(.kirk-button-primary:active) {
    background-color: ${color.primaryActive};
  }

  :global(.kirk-button-primary:focus) {
    box-shadow: 0 0 0 0.1rem ${transparentize(0.75, color.primaryActive)};
    transition: 0.2s box-shadow;
  }

  :global(.kirk-button-secondary) {
    background-color: ${color.white};
    border-color: ${color.border};
    color: ${color.accent};
  }
  :global(.kirk-button-secondary.kirk-button-shadowed) {
    border: none;
  }

  :global(.kirk-button-secondary:hover),
  :global(.kirk-button-secondary:active) {
    background-color: ${color.secondaryActive};
  }

  :global(.kirk-button-secondary:focus) {
    box-shadow: 0 0 0 0.1rem ${transparentize(0.75, color.primaryActive)};
    transition: 0.2s box-shadow;
  }

  :global(.kirk-button-loading) {
    background-color: transparent;
    max-width: initial;
    border: 0;
  }

  :global(.kirk-button-loading:hover),
  :global(.kirk-button-loading:focus),
  :global(.kirk-button-loading:active) {
    background-color: transparent;
  }

  :global(.kirk-button-loading[disabled]) {
    opacity: 1;
  }

  :global(.kirk-button-loading span) {
    opacity: 0;
  }

  :global(svg.kirk-button-loader) {
    position: absolute;
    top: 0;
    left: 50%;
    margin-left: -24px;
  }

  :global(.kirk-button-checked) {
    border: 0;
  }

  :global(.kirk-button-checked[disabled]) {
    opacity: 1;
  }

  :global(.kirk-button-checked span) {
    opacity: 0;
  }

  :global(.kirk-button-warning) {
    background-color: ${color.danger};
    color: ${color.white};
  }

  :global(.kirk-button-icon) {
    display: inline-block;
    padding: 0;
    line-height: 0;
    width: ${buttonIconSize};
    height: ${buttonIconSize};
  }

  /* Note: Safari 10 can not have flex context on <button> element */
  :global(.kirk-button-icon span) {
    display: flex;
    align-items: center;
    justify-content: center;
    width: ${buttonIconSize};
    height: ${buttonIconSize};
  }

  :global(.kirk-button-icon.kirk-button-shadowed span) {
    width: calc(${buttonIconSize} - ${borderSize} * 2);
    height: calc(${buttonIconSize} - ${borderSize} * 2);
  }

  @media (hover: none), (hover: on-demand) {
    :global(.kirk-button-secondary:hover) {
      background-color: ${color.white};
    }
  }
`
