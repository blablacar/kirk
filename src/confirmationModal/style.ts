import css from 'styled-jsx/css'
import { color, space, font } from '_utils/branding'

const footerHeight = '96px' /* = padding + content */

export default css`
  :global(.kirk-modal.kirk-confirmationModal) {
    background-color: rgba(5, 71, 82, 0.95);
    color: ${color.white};
  }

  :global(.kirk-modal.kirk-confirmationModal .kirk-modal-dialog) {
    background: transparent;
    width: auto !important;
    padding: 0;
    max-width: initial;
    margin: 0;
    display: flex;
    height: 100%;
  }

  :global(.kirk-modal.kirk-confirmationModal--large) {
    text-align: center;
  }

  :global(.kirk-modal .kirk-confirmationModal-dialog) {
    position: relative;
    margin: 0 auto;
    box-shadow: none;
    width: auto;
    min-height: 100%;
    padding-top: ${space.xl};
    padding-bottom: ${footerHeight};
  }

  :global(.kirk-modal.kirk-confirmationModal--large .kirk-confirmationModal-dialog) {
    max-width: 600px;
  }

  :global(.kirk-modal .kirk-confirmationModal-body) {
    padding: ${space.xl};
    font-size: ${font.xl.size};
    line-height: ${font.xl.lineHeight};
  }

  :global(.kirk-modal .kirk-confirmationModal-icon) {
    margin-top: ${space.xl};
    margin-left: ${space.xl};
  }

  :global(.kirk-modal .kirk-confirmationModal-footer) {
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: ${footerHeight};
    display: flex;
    justify-content: flex-end;
    padding: ${space.xl};
    background-color: ${color.warningBackground};
  }

  :global(.kirk-modal.kirk-confirmationModal--large .kirk-confirmationModal-footer) {
    position: relative;
    display: block;
    margin: 0 auto;
    background: none;
  }

  :global(.kirk-modal .kirk-confirmationModal-footer) {
    justify-content: flex-end;
  }

  :global(.kirk-modal.kirk-confirmationModal--hasCloseButton .kirk-confirmationModal-footer) {
    justify-content: space-between;
  }

  :global(.kirk-modal.kirk-confirmationModal--large .kirk-confirmationModal-footer .kirk-button) {
    margin: ${space.l} ${space.xl} 0;
  }

  :global(.kirk-modal .kirk-confirmationModal-closeButton) {
    vertical-align: bottom; /* vertical alignment to be fixed in Button */
    border: 0; /* content alignment to be fixed in Button */
  }
`
