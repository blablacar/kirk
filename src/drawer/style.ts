import css from 'styled-jsx/css'
import { color, transition } from '_utils/branding'

export default css`
  .drawer {
    position: fixed;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    z-index: 1;
  }

  :global(.scroll-lock) {
    overflow-y: hidden;
  }

  .drawer::after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    bottom: 0;
    right: 0;
    z-index: 1;
    background-color: rgba(0, 0, 0, 0);
    transition: background-color 150ms linear;
  }

  .drawer--open::after {
    background-color: rgba(0, 0, 0, .3);
  }

  .drawer--close::after {
    background-color: rgba(0, 0, 0, 0);
  }

  .scrollableContent {
    position: absolute;
    z-index: 1050;
    top: 0;
    left: 0;
    bottom: 0;
    overflow-y: auto;
    background-color: ${color.white};
    box-shadow: 0 36px 36px 0 rgba(0, 0, 0, .3);
    transition: transform ${transition.duration.base} ease-in-out;
    will-change: transform;
    transform: translateX(-100%);
    max-width: 100%;
    -webkit-overflow-scrolling: touch;
  }

  .drawer--open .scrollableContent {
    transform: translateX(0%);
  }

  .drawer--close .scrollableContent {
    transform: translateX(-100%);
  }
`
