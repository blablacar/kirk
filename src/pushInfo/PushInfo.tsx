import React, { Component } from 'react'
import cc from 'classcat'

import { animationDelay, animationDuration, StyledPushInfo } from './PushInfo.style'

export interface PushInfoProps {
  readonly className?: string
  readonly icon?: React.ReactNode
  readonly headline: string
  readonly content?: string
  readonly onAnimationEnd?: Function
}

export class PushInfo extends Component<PushInfoProps> {
  componentDidMount() {
    if (this.props.onAnimationEnd) {
      setTimeout(this.props.onAnimationEnd, animationDuration + animationDelay)
    }
  }

  render() {
    const { className, icon, headline, content } = this.props
    return (
      <StyledPushInfo className={cc(['kirk-pushInfo', className])}>
        {icon && <div className="kirk-pushInfo-icon">{icon}</div>}
        <div>
          <h2
            className={cc([
              'kirk-pushInfo-headline',
              { 'kirk-pushInfo-headline--standalone': !content },
            ])}
          >
            {headline}
          </h2>
          {content && <p className="kirk-pushInfo-content">{content}</p>}
        </div>
      </StyledPushInfo>
    )
  }
}
