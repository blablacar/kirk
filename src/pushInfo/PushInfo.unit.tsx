import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import { ProximityIcon } from '../icon/proximityIcon'
import { PushInfo } from './index'
import { animationDelay, animationDuration } from './PushInfo.style'

it('Should have the correct attributes and text.', () => {
  const pushInfo = renderer
    .create(
      <PushInfo
        headline="If it's green it's a win!"
        content="Green icons show meeting points closest to you!"
      />,
    )
    .toJSON()
  expect(pushInfo).toMatchSnapshot()
})

it('Should also have the correct icon.', () => {
  const pushInfo = renderer
    .create(
      <PushInfo
        icon={<ProximityIcon title="" />}
        headline="If it's green it's a win!"
        content="Green icons show meeting points closest to you!"
      />,
    )
    .toJSON()
  expect(pushInfo).toMatchSnapshot()
})

it('Should call the onAnimationEnd prop after some time', () => {
  jest.useFakeTimers()
  const onAnimationEnd = jest.fn()
  mount(
    <PushInfo
      headline="If it's green it's a win!"
      content="Green icons show meeting points closest to you!"
      onAnimationEnd={onAnimationEnd}
    />,
  )

  expect(setTimeout).toHaveBeenCalledTimes(1)
  expect(setTimeout).toHaveBeenLastCalledWith(onAnimationEnd, animationDuration + animationDelay)
})
