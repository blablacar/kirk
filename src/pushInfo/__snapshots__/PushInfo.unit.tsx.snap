// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should also have the correct icon. 1`] = `
.c1 {
  stroke: #5DD167;
}

.c1.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c1.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c1 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #054752;
  color: #FFF;
  border-radius: 16px;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  padding: 0 16px;
  max-height: 0;
  overflow-y: hidden;
  -webkit-animation: slide-down 700ms ease-out 300ms forwards;
  animation: slide-down 700ms ease-out 300ms forwards;
}

.c0 .kirk-pushInfo-headline {
  font-size: 18px;
  line-height: 20px;
  font-weight: 500;
  margin: 0 0 8px;
}

.c0 .kirk-pushInfo-headline--standalone {
  margin: 0;
}

.c0 .kirk-pushInfo-content {
  margin: 0;
}

.c0 .kirk-pushInfo-icon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  min-width: 40px;
  height: 40px;
  margin-right: 16px;
  background-color: #FFF;
}

<div
  className="kirk-pushInfo c0"
>
  <div
    className="kirk-pushInfo-icon"
  >
    <svg
      aria-hidden={true}
      className="kirk-icon c1"
      fill="#5DD167"
      height={24}
      viewBox="0 0 20 20"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      
      <path
        d="M10 20C4.477 20 0 15.523 0 10S4.477 0 10 0s10 4.477 10 10-4.477 10-10 10zm1.202-14.863a1.568 1.568 0 1 0 0-3.136 1.568 1.568 0 0 0 0 3.136zm.94 1.868h-.005a1.568 1.568 0 0 0-1.563-1.554h-.007c-2.252 0-3.896 1.32-4.288 2.692L5.5 11.051H6.926l.723-2.72 1.166-.825-.874 3.264L6.003 18h1.569l1.83-5.661 1.526 1.269.314 4.392h1.528v-5.02l-1.757-1.756.473-1.767a3.753 3.753 0 0 0 2.853 1.313h.314V9.515a2.51 2.51 0 0 1-2.51-2.51z"
        fill="#5DD167"
      />
    </svg>
  </div>
  <div>
    <h2
      className="kirk-pushInfo-headline"
    >
      If it's green it's a win!
    </h2>
    <p
      className="kirk-pushInfo-content"
    >
      Green icons show meeting points closest to you!
    </p>
  </div>
</div>
`;

exports[`Should have the correct attributes and text. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #054752;
  color: #FFF;
  border-radius: 16px;
  font-size: 16px;
  line-height: 20px;
  font-weight: 400;
  padding: 0 16px;
  max-height: 0;
  overflow-y: hidden;
  -webkit-animation: slide-down 700ms ease-out 300ms forwards;
  animation: slide-down 700ms ease-out 300ms forwards;
}

.c0 .kirk-pushInfo-headline {
  font-size: 18px;
  line-height: 20px;
  font-weight: 500;
  margin: 0 0 8px;
}

.c0 .kirk-pushInfo-headline--standalone {
  margin: 0;
}

.c0 .kirk-pushInfo-content {
  margin: 0;
}

.c0 .kirk-pushInfo-icon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 50%;
  min-width: 40px;
  height: 40px;
  margin-right: 16px;
  background-color: #FFF;
}

<div
  className="kirk-pushInfo c0"
>
  <div>
    <h2
      className="kirk-pushInfo-headline"
    >
      If it's green it's a win!
    </h2>
    <p
      className="kirk-pushInfo-content"
    >
      Green icons show meeting points closest to you!
    </p>
  </div>
</div>
`;
